io/src/test/scala/sbt/io/IOSpec.scala

```diff
@@ -70,6 +70,101 @@ class IOSpec extends FunSuite {
     assert(IO.getModifiedTimeOrZero(file("/not/existing/path")) == 0L)
   }
 
+  // classLocation -----------
+
+  test("classLocation[Integer] should return a URL pointing to the rt.jar or java.base") {
+    val u = IO.classLocation[java.lang.Integer]
+    assert(
+      (u.toString == "jrt:/java.base") ||
+        (u.toString.startsWith("file:") && u.toString.endsWith("rt.jar") && IO.asFile(u).isFile))
+  }
+
+  test(
+    "classLocation[AbstractMap.SimpleEntry] should return a URL pointing to the rt.jar or java.base") {
+    val u = IO.classLocation[java.util.AbstractMap.SimpleEntry[String, String]]
+    assert(
+      (u.toString == "jrt:/java.base") ||
+        (u.toString.startsWith("file:") && u.toString.endsWith("rt.jar") && IO.asFile(u).isFile))
+  }
+
+  test("classLocation[this.type] should return a URL pointing to a directory") {
+    val u = IO.classLocation[this.type]
+    assert(IO.asFile(u).isDirectory)
+  }
+
+  // classLocationPath -----------
+
+  test("classLocationPath[Integer] should return NIO path pointing to the rt.jar or java.base") {
+    val p = IO.classLocationPath[java.lang.Integer]
+    assert(
+      ((p.toString == "/modules/java.base") && (p.getFileSystem.toString == "jrt:/"))
+        || (p.toString.endsWith("rt.jar") && p.toFile.isFile)
+    )
+  }
+
+  test(
+    "classLocationPath[AbstractMap.SimpleEntry] should return NIO path pointing to the rt.jar or java.base") {
+    val p = IO.classLocationPath[java.util.AbstractMap.SimpleEntry[String, String]]
+    assert(
+      ((p.toString == "/modules/java.base") && (p.getFileSystem.toString == "jrt:/"))
+        || (p.toString.endsWith("rt.jar") && p.toFile.isFile)
+    )
+  }
+
+  test("classLocationPath[this.type] should return NIO path pointing to a directory") {
+    val p = IO.classLocationPath[this.type]
+    assert(p.toFile.isDirectory)
+  }
+
+  // classLocationFileOption -----------
+
+  test("classLocationFileOption[Integer] should return File pointing to the rt.jar or None") {
+    val opt = IO.classLocationFileOption[java.lang.Integer]
+    assert(
+      opt match {
+        case Some(x) => x.getName == "rt.jar"
```

Comment:
Can we add at least one jdk11 build to .travis.yml to ensure that both branches are executed during ci?

Reply:
Yes. That was the main purpose of #181.

Reply:
https://travis-ci.org/sbt/io/jobs/436985929